keps/sig-auth/3766-referencegrant/README.md

```diff
@@ -0,0 +1,906 @@
+
+The new version will look like this instead:
+```yaml
+apiVersion: authorization.k8s.io/v1alpha1
+kind: ReferenceGrant
+metadata:
+  name: allow-gateways
+  namespace: bar
+spec:
+  from:
+    # Assuming that we leave the default for Group to the empty
+    # string, so that Core objects don't need additional config.
+    - group: "gateway.networking.kubernetes.io" 
+      resource: gateways
+      namespace: foo
+  to:
+    - resource: secrets
+
+```
+
+The new version communicates the scope more clearly because `resource` is
+unambiguous and corresponds to exactly one URL path on the API Server.
+
+This change also leaves room for an enhancement. Whether we have an in-tree or
+CRD implementation, we can rely on the exact matching that the plural resource
+name gives us, and [warn](https://kubernetes.io/blog/2020/09/03/warnings/) if
+either side of the grant is for an API that's not served by this cluster.
+
+#### Revocation behavior
+
+Unfortunately, there's no way to be specific about what happens when a
+ReferenceGrant is deleted in every possible case - the revocation behavior is
+dependent on what access is being granted (and revoked). With that said, we
+expect the following guidelines to be rules to apply to ALL implementations of
+the API:
+
+* Deletion of a ReferenceGrant means the granted access is revoked.
+* ReferenceGrant controllers must remove any configuration generated by the
+  granted access as soon as possible (eventual consistence permitting).
+* Some actions that have already been enabled by the ReferenceGrant (such as
+  forwarding requests or persisting data) cannot be undone, but no future
+  actions should be allowed.
+
+The examples below include information about what happens when the ReferenceGrant
+is removed as data points.
+
+### Example Usage
+
+#### Gateway API Gateway Referencing Secret
+
+In this example (from the Gateway API docs), we have a Gateway in the
+`gateway-api-example-ns1` namespace, referencing a Secret in the
+`gateway-api-example-ns2` namespace. The following ReferenceGrant allows this:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: cross-namespace-tls-gateway
+  namespace: gateway-api-example-ns1
+spec:
+  gatewayClassName: acme-lb
+  listeners:
+  - name: https
+    protocol: HTTPS
+    port: 443
+    hostname: "*.example.com"
+    tls:
+      # There's a Kind/Resource mismatch here, which sucks, but it is not
+      # easily fixable, since Gateway is already a beta, close to GA
+      # object.
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: wildcard-example-com-cert
+        namespace: gateway-api-example-ns2
+---
+apiVersion: authorization.k8s.io/v1alpha1
+kind: ReferenceGrant
+metadata:
+  name: allow-ns1-gateways-to-ref-secrets
+  namespace: gateway-api-example-ns2
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    resource: gateways
+    namespace: gateway-api-example-ns1
+  to:
+  - resource: secrets
```

Comment:
let's err on the side of keeping group explicit in the examples for now

```suggestion
  - group: ""
    resource: secrets
```

Reply:
Updated, thanks!